components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1969,13 +2015,56 @@ public String addSubscription(APIIdentifier identifier, String userId, int appli
 
     public void removeSubscription(APIIdentifier identifier, String userId, int applicationId)
             throws APIManagementException {
-        apiMgtDAO.removeSubscription(identifier, applicationId);
+        API api = getAPI(identifier);
+        boolean isTenantFlowStarted = false;
+
+        if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+            PrivilegedCarbonContext.startTenantFlow();
+            isTenantFlowStarted = true;
```

Comment:
Please check the comment [isTenantFlowStarted](https://github.com/wso2/carbon-apimgt/pull/1094#discussion_r57555401)